src/catalog/pgduckdb_transaction_manager.cpp

```diff
@@ -1,15 +1,21 @@
 #include "pgduckdb/catalog/pgduckdb_transaction_manager.hpp"
 #include "duckdb/main/attached_database.hpp"
+#include "pgduckdb/pgduckdb_process_lock.hpp"
 
 namespace duckdb {
 
-PostgresTransactionManager::PostgresTransactionManager(AttachedDatabase &db_p, PostgresCatalog &catalog,
-                                                       Snapshot snapshot)
-    : TransactionManager(db_p), catalog(catalog), snapshot(snapshot) {
+PostgresTransactionManager::PostgresTransactionManager(AttachedDatabase &db_p, PostgresCatalog &catalog)
+    : TransactionManager(db_p), catalog(catalog) {
 }
 
 Transaction &
 PostgresTransactionManager::StartTransaction(ClientContext &context) {
+	Snapshot snapshot;
+	{
+		std::lock_guard<std::mutex> lock(pgduckdb::DuckdbProcessLock::GetLock());
```

Comment:
Does it need the lock? I thought that snapshots were transaction-local, created by postgres when a transaction started and GetActiveSnapshot just references that cached instance?

Or is it created lazily and that's why we need to protect the catalog from getting modified?
I would think postgres has their own locks to protect against this, we're not changing any global data in the shared duckdb db right?

Reply:
Agreed, the lock should is not necessary. GetActiveSnapshot only does a read, so no need to serialize those. There are ofcourse functions that change the activesnapshot, but we don't call those functions.